### PR TITLE
add config for job concurrency count

### PR DESCRIPTION
**Pull request by hongyh13**
add configuration for job concurrency count

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationValidators/ConfigurationValidatorV1.cs [hunk 1]
```diff
@@ -83,6 +83,16 @@ public static void Validate(IServiceCollection services)
                 throw new ConfigurationErrorException($"Target azure container name can not be empty.");
             }
 
+            if (jobConfiguration.MaxConcurrencyCount < 1)
```
Comment (sowu880) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationValidators/ConfigurationValidatorV1.cs:86]: Do we need to add a const variable for maximum count limitation?
(The value could be set after performance tests.)

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationValidators/ConfigurationValidatorV1.cs [hunk 2]
```diff
@@ -83,6 +83,16 @@ public static void Validate(IServiceCollection services)
...
+            {
+                throw new ConfigurationErrorException("Max concurrency count can not less than 1.");
```
[the remark this answers is not in the document]
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationValidators/ConfigurationValidatorV1.cs:88]: Updated

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/JobConfiguration.cs [hunk 3]
```diff
@@ -72,5 +72,17 @@ public class JobConfiguration
         // TODO: don't expose to PaaS
         [JsonProperty("endTime")]
         public DateTimeOffset? EndTime { get; set; }
+
+        /// <summary>
+        /// Max concurrency count in an instance.
+        /// </summary>
+        [JsonProperty("maxConcurrencyCount")]
```
[the remark this answers is not in the document]
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/JobConfiguration.cs:79]: OK, modified.

**Review by moria97 — APPROVED** [review]

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationValidators/ConfigurationValidatorV1.cs (hunk 1 above)
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationValidators/ConfigurationValidatorV1.cs]: I think I need to give the default count after performance tests. The maximum count depends on the target machine, and I think we may don't need to limit it.

Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationValidators/ConfigurationValidatorV1.cs]: Per our offline discussion, we don't need to limit the maximum count here.

**Pull request merged**